src/main/java/io/appform/dropwizard/discovery/bundle/id/IdGenerator.java

```diff
@@ -150,8 +179,10 @@ public static Id generate(final String prefix,
      * @param domain Domain for constraint selection
      * @return Return generated id or empty if it was impossible to satisfy constraints and generate
      */
-    public static Optional<Id> generateWithConstraints(String prefix, String domain) {
-        return generateWithConstraints(prefix, DOMAIN_SPECIFIC_CONSTRAINTS.getOrDefault(domain, Collections.emptyList()), true);
+    public static Optional<Id> generateWithConstraints(String prefix, @NonNull String domain) {
```

Comment:
call the other method with default value of skipGlobal

Reply:
done